### PR TITLE
feat: [SRM-9902]: Persist metric dashboard selection

### PR DESCRIPTION
**Pull request by deepesh-ui**
##### Screenshots:
Screenshot 2022-04-01 at 9 47 19 AM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by harshil562 — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx [hunk 1]
```diff
@@ -80,6 +80,12 @@ function DefineHealthSource(props: DefineHealthSourceProps): JSX.Element {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [])
 
+  // useEffect(() => {
+  //   if (sourceData.selectedDashboards) {
+  //     sourceData.selectedDashboards = new Map()
+  //   }
+  // }, [sourceData?.connectorRef])
```
Comment (shashidhargg) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx:87]: Is it required?

**Review by harshilbits — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx [hunk 2]
```diff
@@ -94,7 +100,11 @@ function DefineHealthSource(props: DefineHealthSourceProps): JSX.Element {
         validationSchema={validate(getString)}
         onSubmit={values => {
           onSubmit?.(values)
-          onNext(values, { tabStatus: 'SUCCESS' })
+          let formValues = values
+          if (sourceData.selectedDashboards && values?.connectorRef !== sourceData?.connectorRef) {
+            formValues = { ...values, selectedDashboards: new Map() }
```
Comment (shashidhargg) [inline src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx:105]: Here we are not using `selectedDashboards`. Instead, we are setting it as empty. Is it correct?

**Review by shashidhargg — APPROVED** [review]

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx:87]: my bad , strange wasn't picked up by Sonar

File: src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/HealthSourceDrawer/component/defineHealthSource/DefineHealthSource.tsx:105]: Yes , this is for a bug https://harness.atlassian.net/browse/SRM-9235,  have refactored code and moved here as we have metrics for Both GCO and DataDog

**Pull request merged**